### ISSUE TITLE
Verification required when adding scope 

### ISSUE THREAD

**Issue by taaariq** [abhinavminhas/Gmail-API-Helper#84]
![image](https://github.com/user-attachments/assets/5124a62a-f2f7-4d38-ba51-acd9cc04d0c2)
![image](https://github.com/user-attachments/assets/306f8317-3c26-4d9c-a14c-f56dc97f4dbd)

https://support.google.com/cloud/answer/13463073?visit_id=638602395330448335-2251483393&rd=1

Any workaround?

**Comment by abhinavminhas** [comment]
You can follow the [configuration](https://github.com/abhinavminhas/Gmail-API-Helper?tab=readme-ov-file#user-content-configuration) steps to set up an external user app. The required scope can work with an unverified app and can be verified later if needed.

**Comment by taaariq** [comment]
I have added `Gmail API https://mail.google.com/ scope` mentioned in the docs and I was able to obtain the token.json folder. However when I try to retrieve emails I get the following error 

```
Unhandled exception. The service gmail has thrown an exception.
HttpStatusCode is Forbidden.
Google.Apis.Requests.RequestError
Metadata scope does not support 'q' parameter [403]
Errors [
        Message[Metadata scope does not support 'q' parameter] Location[ - ] Reason[forbidden] Domain[global]
]

Google.GoogleApiException: The service gmail has thrown an exception. HttpStatusCode is Forbidden. Metadata scope does not support 'q' parameter
   at Google.Apis.Requests.ClientServiceRequest`1.ParseResponse(HttpResponseMessage response)
   at Google.Apis.Requests.ClientServiceRequest`1.Execute()
   at GmailAPIHelper.GmailHelper.GetMessages(GmailService gmailService, String query, Boolean markRead, String userId, Boolean disposeGmailService)
   at Gmail.GetMessage() in /Users/ta/RiderProjects/GmailHelper/gmhelper/Program.cs:line 22
   at Program.<Main>$(String[] args) in /Users/ta/RiderProjects/GmailHelper/gmhelper/Program.cs:line 7

```

My code is as below

```
using GmailAPIHelper;

// Console.WriteLine("Attempting to create o auth token..."); 
// GmailHelper.GetGmailService("xxx-gmail-helper"); //obtain token.json
Gmail.GetMessage();


public class Gmail {

    public static void GetMessage()
    {
        string TestEmailId = "no-reply@accounts.google.com";
        var credPath =Environment.CurrentDirectory + "/" + "token.json";
        Console.Write($"cred path: {credPath}");
        var service = GmailHelper.GetGmailService("xxx-gmail-helper", GmailHelper.TokenPathType.CUSTOM, credPath);

        var messages = GmailHelper.GetMessages(service, query: $"[from:{TestEmailId}]in:inbox is:read");
        Console.Write(messages.ToString());
    }
    
    
}


```

**Comment by hydrojumbo** [comment]
> Metadata scope does not support 'q' parameter [403]

I got the same after following the [configuration steps](https://github.com/abhinavminhas/Gmail-API-Helper?tab=readme-ov-file#user-content-configuration). The solution was to eliminate the `GmailService.Scope.GmailMetadata` scope in the list of scopes being requested when the `GmailService` is instantiated.

To do this, I wrote a helper method that constructs the GmailService slightly differently. Note that I am also specifying specific paths for `credentials.json` so that I can run this in LinqPAD.
